lineapy/graph_reader/node_collection.py

```diff
@@ -99,12 +100,12 @@ class NodeCollection:
     is_pre_computed_artifact: bool = field(default=False)
 
     def __post_init__(self):
-        self.safename = self.name.replace(" ", "_")
+        self.safename = slugify(self.name)
```

Comment:
@mingjerli: I am updating this part as per suggestion by @lionsardesai. I see `self.safename` used only for generating function names, so this should not create an issue.